src/InstallerHosts/Elastic.InstallerHosts/Elasticsearch/Tasks/Install/SetupXPackPasswordsTask.cs

```diff
@@ -111,5 +119,63 @@ private void SetPassword(HttpClient client, string elasticUserPassword, string u
 			}
 			this.Session.SendProgress(200, $"Changed password for user '{user}'");
 		}
+
+		// Might be moved and refactored when is needed for other tasks
+		public static string GetBaseAddress(string networkHost, int? httpPort)
+		{
+			var port = httpPort ?? DefaultHttpPort;
+
+			if (string.IsNullOrEmpty(networkHost))
+				return $"http://localhost:{port}/";
+
+			var host = networkHost;
+			if (networkHost.IndexOf(',') >= 0) // Uri host must not contain comma, but elasticsearch.yml supports it
+			{
+				var hosts = networkHost.Split(new[] {','}, StringSplitOptions.RemoveEmptyEntries).Select(h => h.Trim());
+				host = hosts.FirstOrDefault(h => h == "localhost" || h == "127.0.0.1"
+					|| h == LocalhostPlaceholder || h == LocalhostV4Placeholder || h == LocalhostV6Placeholder) // prefer local interface
+					?? hosts.First();
+			}
+
+			switch (host)
+			{
+				case LocalhostPlaceholder:
+					host = "localhost";
+					break;
+				case LocalhostV4Placeholder:
+					host = "127.0.0.1";
+					break;
+				case LocalhostV6Placeholder:
+					host = "[::1]";
+					break;
+				//TODO: differentiate between _site_ and _global_
+				case SiteAddressPlaceholder:
+				case GlobalAddressPlaceholder:
+				case SiteAddressV4Placeholder:
+				case GlobalAddressV4Placeholder:
+				case SiteAddressV6Placeholder:
+				case GlobalAddressV6Placeholder:
+					IPHostEntry ipHostEntry = Dns.GetHostEntry(Dns.GetHostName());
```

Comment:
Wrap this in an try catch might be safer with a fallback to localhost?

Reply:
This code is to address _global_ placeholder which by definition should not use local interface. Having this in elasticsearch.yml will tell elastic server not to listen locally. So using localhost as a fallback does not make much sense to me. My idea was the following: if _global_ is used resolve of current host name should work or installation should fail with clear reason.

Reply:
You are correct, the preference for local if present is handled earlier.